app/src/ui/toolbar/branch-dropdown.tsx

```diff
@@ -338,6 +341,26 @@ export class BranchDropdown extends React.Component<IBranchDropdownProps> {
     })
   }
 
+  private onViewBranchOnGitHub = () => {
+    const tip = this.props.repositoryState.branchesState.tip
+    const gitHubRepository = this.props.repository.gitHubRepository
+
+    if (tip.kind !== TipState.Valid) {
+      return
+    }
+
+    if (!gitHubRepository || gitHubRepository.htmlURL === null) {
+      return
+    }
+
+    const branchName = tip.branch.upstreamWithoutRemote ?? tip.branch.name
```

Comment:
Since we won't show the context menu item unless `upstreamWithourRemote` is set we should bail here if it isn't as well.

```suggestion
    if (!tip.branch.upstreamWithoutRemote) {
      return
    }

    const url = `${gitHubRepository.htmlURL}/tree/${encodeURIComponent(
      tip.branch.upstreamWithoutRemote
    )}`
```